src/main/java/wolox/training/controllers/BookController.java

```diff
@@ -151,11 +151,11 @@ public List findByAuthor(@PathVariable String bookAuthor) {
             .body(openLibraryService.saveBook(openLibraryBook));
     }
 
-    @GetMapping()
-    @RequestMapping(params = {"publisher", "genre", "year"})
+    @GetMapping("/search")
     public List<Book> findByPublisherAndGenreAndYear(
-        @RequestParam(name = "publisher") String publisher,
-        @RequestParam(name = "genre") String genre, @RequestParam(name = "year") String year) {
+        @RequestParam(name = "publisher", required = false) String publisher,
+        @RequestParam(name = "genre", required = false) String genre,
+        @RequestParam(name = "year", required = false) String year) {
```

Comment:
I think you are losing your previous changes (The modification to have a more restful method name)

Reply:
I mean, what you can do is this to accomplish the restful naming convention:
```
@GetMapping
@RequestMapping(params = {"publisher", "genre", "year"})
```